moto/sns/models.py

```diff
@@ -257,7 +257,7 @@ def list_subscriptions(self, topic_arn=None, next_token=None):
             return self._get_values_nexttoken(self.subscriptions, next_token)
 
     def publish(self, arn, message, subject=None):
-        if subject is not None and len(subject) > 100:
+        if subject is not None and len(subject) >= 100:
```

Comment:
It would be even-better-if the supplied message was copied into the publish call, so it can round trip. Right now, any message always contains the default ``"my subject"`` from ``moto/sns/models.py``.

Since there may be more optional fields, maybe "pass all" might be a reasonable approach. E.g:
```python
    def publish(self, arn, message, subject=None, **kwargs):
...
        if subject is not None:
            kwargs['subject'] = subject
            if len(subject) >= 100:
...
        message_id = topic.publish(message, **kwargs)
```

Reply:
nm -- the current functionality helps a lot -- adding roundtrip support looks more complex than I thought. It should be a separate bug after this lands, I think.

Reply:
^^ Agreed.